src/marten/cli/generator/auth/templates/app/handlers/password_update_handler.cr.ecr

```diff
@@ -0,0 +1,28 @@
+require "./concerns/*"
+
+module <%= context.module_name %>
+  class PasswordUpdateHandler < Marten::Handlers::Schema
+    include RequireSignedInUser
+
+    schema PasswordUpdateSchema
+    template_name "<%= context.label %>/password_update.html"
+    success_route_name "<%= context.label %>:profile"
+
+    before_schema_validation :prepare_schema
+    after_successful_schema_validation :process_password_reset_request
+
+    private def prepare_schema
+      schema.email = request.user!.email
```

Comment:
Since the user is already authenticated, couldn't we make use of this in order to assign the current user to the schema instead of its email only? If we do this, this will allow us to avoid re-retrieving the user record from the DB (with `MartenAuth#authenticate`) and instead, we will be able to verify the old password by simply calling `User#check_password`.